build/xtask/src/main.rs

```diff
@@ -121,34 +116,29 @@ fn build(sh: &Shell, rl: bool, mut args: Arguments) -> Result<()> {
     let build_dir = if rl { "release" } else { "debug" };
 
     let diskname = "xernel.hdd";
-    let disksize = 64.to_string();
 
-    cmd!(
-        sh,
-        "dd if=/dev/zero of={diskname} bs=1M count=0 seek={disksize}"
-    )
-    .run()?;
+    let data_vec = vec![0_u8; 64 * 1024 * 1024];
+    let mut disk = Cursor::new(data_vec);
 
-    cmd!(sh, "mformat -i {diskname} -F").run()?;
-    cmd!(
-        sh,
-        "mcopy -i {diskname} ./target/{target}/{build_dir}/xernel ::/xernel"
-    )
-    .run()?;
-    cmd!(
-        sh,
-        "mcopy -i {diskname} xernel/kernel/limine.cfg ::/limine.cfg"
-    )
-    .run()?;
+    format_volume(&mut disk, FormatVolumeOptions::new().fat_type(fatfs::FatType::Fat32))?;
 
-    cmd!(sh, "mcopy -i {diskname} ./logo.bmp ::/logo.bmp").run()?;
-    cmd!(sh, "mmd -i {diskname} ::/EFI").run()?;
-    cmd!(sh, "mmd -i {diskname} ::/EFI/BOOT").run()?;
-    cmd!(
-        sh,
-        "mcopy -i {diskname} xernel/kernel/limine/BOOTX64.EFI ::/EFI/BOOT"
-    )
-    .run()?;
+    let fs = fatfs::FileSystem::new(&mut disk, fatfs::FsOptions::new())?;
+    {
+        let root_dir = fs.root_dir();
+
+        copy_to_image(&root_dir, &format!("./target/{target}/{build_dir}/xernel"), "xernel")?;
+
+        copy_to_image(&root_dir, "./xernel/kernel/limine.cfg", "limine.cfg")?;
```

Comment:
I would put the limine configuration file in the same folder as the limine bootloader